packages/react-devtools-shared/src/hook.js

```diff
@@ -339,5 +339,22 @@ export function installHook(target: any): DevToolsHook | null {
     }: Object),
   );
 
+  target.__REACT_DEVTOOLS_GLOBAL_HOOK__.on('renderer', function(evt) {
+    target.postMessage(
+      {
+        source: 'react-devtools-detector',
+        reactBuildType: evt.reactBuildType,
+      },
+      '*',
+    );
+  });
+
+  target.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeObjectCreate = Object.create;
+  target.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeMap = Map;
+  target.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeWeakMap = WeakMap;
+  target.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeSet = Set;
+
   return hook;
 }
+
+installHook(window);
```

Comment:
this line is a bit sketchy there are [multiple places](https://github.com/facebook/react/search?q=installHook) where this file is imported, depending on how the importing works if the whole file is imported it will get called with window immediatelly. it looks like most of these files are calling this function similarly ( immediatelly ) [except for the backend](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react-devtools-inline/src/backend.js#L105).

An alternate approach that could give us some saftey is to back out the changes to this hook file then create an extension hook file that will import this hook and then apply the changes that were made here, something like

```javascript
// extension-hook.js
import {installHook} from 'react-devtools-shared/src/hook';

installHook(window);

window.__REACT_DEVTOOLS_GLOBAL_HOOK__.on('renderer', function(evt) {
  window.postMessage(
    {
      source: 'react-devtools-detector',
      reactBuildType: evt.reactBuildType,
    },
    '*',
  );
});

window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeObjectCreate = Object.create;
window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeMap = Map;
window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeWeakMap = WeakMap;
window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeSet = Set;

```

that way any other importers of `react-devtools-shared/src/hook.js` will still use the function as it was before without any changes